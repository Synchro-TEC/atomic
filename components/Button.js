import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Button extends Component {
  render() {
    return <button className="sv-button primary" onClick={() => this.props.onClick()}>{this.props.children}</button>;
  }
}

Button.propTypes = {
  theme: PropTypes.string,
}

export default Button;
