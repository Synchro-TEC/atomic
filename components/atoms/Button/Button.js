import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Button = ({ outline, className, theme, onClick, style, children, full, small, ...otherProps }) => {
  const baseClass = 'sv-button';
  const buttonTheme = outline ? `out-${theme}` : theme;
  const rootClass = cx(baseClass, buttonTheme, { full, small }, className);

  return (
    <button className={rootClass} style={style} onClick={onClick} {...otherProps}>
      {children}
    </button>
  );
};

Button.defaultProps = {
  theme: 'default',
};

Button.propTypes = {
  className: PropTypes.string,
  full: PropTypes.bool,
  onClick: PropTypes.func,
  outline: PropTypes.bool,
  small: PropTypes.bool,
  style: PropTypes.object,
  theme: PropTypes.oneOf(['default', 'primary', 'info', 'danger', 'warning']),
};

export default Button;
