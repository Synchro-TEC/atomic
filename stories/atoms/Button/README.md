The `<Button>` component is a button.