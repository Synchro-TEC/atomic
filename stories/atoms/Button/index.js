import React from "react";
import { text, boolean, object } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { optionalSelect } from "../../../utils/optionalSelect";

import readme from "./README.md";
import Button from "../../../components/Button";

const options = {
  default: "default",
  primary: "primary",
  info: "info",
  warning: "warning",
  danger: "danger",
};
const component = () => (
  <Button
    theme={optionalSelect("Theme", options, "")}
    outline={boolean("Outline", false)}
    full={boolean("Full", false)}
    small={boolean("Small", false)}
    className={text("ClassName", "")}
    style={object("Style", {})}
    onClick={action("button_clicked")}
  >
    My Button
  </Button>
);

export default [readme, component];