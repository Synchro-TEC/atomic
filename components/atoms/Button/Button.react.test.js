import React from 'react';
import Button from './Button';

describe('Button Component', () => {
  test('Render the button', () => {
    const wrapper = shallow(<Button>The Button</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  test('Fire onClick button', () => {
    const mockCallBack = jest.fn();
    const button = shallow(<Button onClick={mockCallBack}>The Button</Button>);
    button.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
