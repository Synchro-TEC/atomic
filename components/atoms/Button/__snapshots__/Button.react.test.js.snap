// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Render the button 1`] = `
<button
  className="sv-button default"
>
  The Button
</button>
`;
